### PR TITLE
implemented search mask

### PR DESCRIPTION
**Pull request by SabMueller**
Hey guys!

I've finished my user story on the animal search mask! I added the following: 
- implemented 2 dropdown menus in AnimalSearch.js, to define search criteria
- implemented filter & search logic
- adjusted mapping on Main.js from all animals to filtered animals
- added error handling on Main.js in case there are no results
- added styling

Thank you for your review! ❤️ ✨

### CODE REVIEW

File: client/src/pages/AnimalSearch.js [hunk 1]
```diff
@@ -1,10 +1,211 @@
...
+  span {
+    color: var(--primary-lightest);
+  }
+`;
+
+const TypeWrapper = styled.section`
+  display: grid;
+  place-items: center;
+
+  margin-top: 1.5rem;
```
Comment (bejanaf) [inline client/src/pages/AnimalSearch.js:159]: please sort alphabetically :)

File: client/src/pages/Main.js [hunk 2]
```diff
@@ -21,3 +45,37 @@ const ResultsText = styled.p`
   letter-spacing: 0.05rem;
   padding: 0.5rem;
 `;
+
+const ErrorWrapper = styled.section`
+  letter-spacing: 0.05rem;
+  padding: 4rem;
+  display: grid;
```
Comment (bejanaf) [inline client/src/pages/Main.js:50]: In your structuring you usually have the display flex design and the associated adjustments in the first place :)

**Review by bejanaf — APPROVED** [review]
✨ Awesome, looks great! :)

File: client/src/pages/Main.js (hunk 2 above)
Comment (SabMueller) [reply client/src/pages/Main.js:50]: thank you! :)

**Pull request merged**